src/modules/logger/logger.cpp

```diff
@@ -492,8 +492,7 @@ void Logger::run()
 						 */
 						message_data_header_s *header = reinterpret_cast<message_data_header_s *>(buffer);
 						header->msg_type = static_cast<uint8_t>(MessageType::DATA);
-						/* the ORB topic data object has 2 unused trailing bytes? */
-						header->msg_size = static_cast<uint16_t>(msg_size - 2);
+						header->msg_size = static_cast<uint16_t>(msg_size - 3);
```

Comment:
The 3 bytes comprising msg_type and msg_size which have already been read
by the time a record is parsed.

On Tue, May 3, 2016 at 9:47 PM Daniel Agar <notifications@github.com> wrote:

> In src/modules/logger/logger.cpp
> <https://github.com/PX4/Firmware/pull/4365#discussion_r61987833>:
>
> > @@ -492,8 +492,7 @@ void Logger::run()
> >  						 */
> >  						message_data_header_s *header = reinterpret_cast<message_data_header_s *>(buffer);
> >  						header->msg_type = static_cast<uint8_t>(MessageType::DATA);
> > -						/* the ORB topic data object has 2 unused trailing bytes? */
> > -						header->msg_size = static_cast<uint16_t>(msg_size - 2);
> > +						header->msg_size = static_cast<uint16_t>(msg_size - 3);
>
> @kd0aij <https://github.com/kd0aij> where does the 3 come from?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/PX4/Firmware/pull/4365/files/13743282c57ca31769ccdaae103bf5a2a70fdcd7..6bbb14eb4fe90e81c1d0ebef4552032a6b93dd93#r61987833>
>